src/daft-plan/src/physical_plan.rs

```diff
@@ -142,46 +134,32 @@ impl PhysicalPlan {
                 descending.clone(),
             ))
             .into(),
-            Self::Split(Split {
-                output_num_partitions,
+            Self::ShuffleExchange(shuffle_exchange) => shuffle_exchange.clustering_spec(),
+            Self::Aggregate(Aggregate {
+                input,
+                aggregations,
                 ..
             }) => {
-                ClusteringSpec::Unknown(UnknownClusteringConfig::new(*output_num_partitions)).into()
-            }
-            Self::Coalesce(Coalesce { num_to, .. }) => {
-                ClusteringSpec::Unknown(UnknownClusteringConfig::new(*num_to)).into()
-            }
-            Self::Flatten(Flatten { input }) => input.clustering_spec(),
-            Self::FanoutRandom(FanoutRandom { num_partitions, .. }) => {
-                ClusteringSpec::Random(RandomClusteringConfig::new(*num_partitions)).into()
-            }
-            Self::FanoutByHash(FanoutByHash {
-                num_partitions,
-                partition_by,
-                ..
-            }) => ClusteringSpec::Hash(HashClusteringConfig::new(
-                *num_partitions,
-                partition_by.clone(),
-            ))
-            .into(),
-            Self::FanoutByRange(FanoutByRange {
-                num_partitions,
-                sort_by,
-                descending,
-                ..
-            }) => ClusteringSpec::Range(RangeClusteringConfig::new(
-                *num_partitions,
-                sort_by.clone(),
-                descending.clone(),
-            ))
-            .into(),
-            Self::ReduceMerge(ReduceMerge { input }) => input.clustering_spec(),
-            Self::Aggregate(Aggregate { input, groupby, .. }) => {
-                let input_clustering_spec = input.clustering_spec();
-                if groupby.is_empty() {
-                    ClusteringSpec::Unknown(Default::default()).into()
```

Comment:
NOTE: The previous implementation seems to be a bug which was masked somehow by the previous implementation of `FanoutMap -> ReduceMerge`. I implemented a fix which I think is appropriate.